### PR TITLE
Adding Tests

### LINKED ISSUE #4: Add tests
Tests should be added to the project to ensure each PR does not introduce breaking changes.

I am currently working on this in the [add-tests](https://github.com/JeffResc/sharkiq/tree/add-tests) branch and will open a PR when they are ready.

### PR DESCRIPTION
**Pull request by travipross**
Closes #4 

This introduces a `pytest` suite of tests along with some configuration. So far, a number of unit tests have been written to cover all of the AylaApi methods that do not make any external web request. More tests can be added to this PR, or a new one could be created in the future to extend this baseline.

I'd also be happy to integrate the tests into a new github actions workflow to run on pull requests, however, I lack the permissions. Proper authentication/integration tests would also likely require a valid Shark login to be stored as a repository secret.

### CODE REVIEW

File: sharkiq/ayla_api.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def __init__(
         self._app_secret = app_secret
         self.websession = websession
 
-    def ensure_session(self) -> aiohttp.ClientSession:
+    async def ensure_session(self) -> aiohttp.ClientSession:
```
Comment (travipross) [inline sharkiq/ayla_api.py:51]: A `DeprecationWarning` from `aiohttp` was being issued due to the `aiohttp.ClientSession` object being created outside of an `async` method. Added this and the below `await`s to address this warning.

File: sharkiq/ayla_api.py [hunk 2]
```diff
@@ -75,7 +75,7 @@ def _set_credentials(self, status_code: int, login_result: Dict):
         self._access_token = login_result["access_token"]
         self._refresh_token = login_result["refresh_token"]
         self._auth_expiration = datetime.now() + timedelta(seconds=login_result["expires_in"])
-        self._is_authed = True
+        self._is_authed = True  # TODO: Any non 200 status code should cause this to be false
```
Comment (travipross) [inline sharkiq/ayla_api.py:78]: This method explicitly checks for two specific error codes and assumes any other response was successful.

File: requirements.test.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+pytest-asyncio-0.18.3
+pytest==7.1.1
+pytest-cov==3.0.0
```
Comment (travipross) [inline requirements.test.txt:3]: Locally I've been using `pytest-cov` to generate a coverage report when executing the tests:
```
pytest --cov=sharkiq --cov-report html --cov-report term
```

**Comment by funkybunch** [comment]
@JeffResc looks good to me but since you had been working more closely with the tests could you take a look as well?

**Comment by JeffResc** [comment]
Hi @travipross - Thanks for the PR, I really appreciate it! I don't have time to review this right this second, but I just wanted to reach out and say thank you. I should have time later this week and definitely by the end of this weekend to give this a look.

**Comment by travipross** [comment]
No rush on my end! I'll probably keep tinkering away until then

File: sharkiq/ayla_api.py (hunk 2 above)
Comment (JeffResc) [reply sharkiq/ayla_api.py:78]: I have moved this to issue #11 for the time being so that I can merge this

File: requirements.test.txt (hunk 3 above)
Comment (JeffResc) [reply requirements.test.txt:3]: Workflow has been added 👍

File: sharkiq/ayla_api.py (hunk 1 above)
Comment (JeffResc) [reply sharkiq/ayla_api.py:51]: LGTM 👍

**Comment by JeffResc** [comment]
We're a little lacking on coverage, but it's definitely a start! Thanks @travipross 👍

**Pull request merged**